homeassistant/components/compit/climate.py

```diff
@@ -0,0 +1,278 @@
+) -> None:
+    """Set up the CompitClimate platform from a config entry."""
+
+    coordinator: CompitDataUpdateCoordinator = entry.runtime_data
+
+    async_add_devices(
+        [
+            CompitClimate(
+                coordinator,
+                device,
+                device_definition.parameters,
+                device_definition.name,
+            )
+            for gates in coordinator.gates
+            for device in gates.devices
+            if (
+                device_definition := next(
+                    (
+                        definition
+                        for definition in coordinator.device_definitions.devices
+                        if definition.code == device.type
+                    ),
+                    None,
+                )
+            )
+            is not None
+            if (device_definition.device_class == 10)
+        ]
+    )
+
+
+class CompitClimate(CoordinatorEntity[CompitDataUpdateCoordinator], ClimateEntity):
+    """Representation of a Compit climate device."""
+
+    def __init__(
+        self,
+        coordinator: CompitDataUpdateCoordinator,
+        device: Device,
+        parameters: list[Parameter],
+        device_name: str,
+    ) -> None:
+        """Initialize the climate device."""
+        super().__init__(coordinator)
+        self.coordinator = coordinator
+        self._attr_unique_id = f"{device.label}_{device.id}"
+        self._attr_name = device.label
+        self._attr_has_entity_name = True
+        self._attr_temperature_unit = UnitOfTemperature.CELSIUS
+        self._attr_hvac_modes = [HVACMode.HEAT, HVACMode.COOL, HVACMode.OFF]
+        self._attr_supported_features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.FAN_MODE
+            | ClimateEntityFeature.PRESET_MODE
+        )
+        self._attr_device_info = DeviceInfo(
+            {
+                "identifiers": {(DOMAIN, str(device.id))},
+                "name": device.label,
+                "manufacturer": MANURFACER_NAME,
+                "model": device_name,
+                "sw_version": "1.0",
+            }
+        )
+
+        parametersDict = {
+            parameter.parameter_code: parameter for parameter in parameters
+        }
+        self.device = device
+        self.available_presets: Parameter | None = parametersDict.get(PARAM_PRESET_MODE)
+        self.available_fan_modes: Parameter | None = parametersDict.get(PARAM_FAN_MODE)
+        self.available_hvac_modes: Parameter | None = parametersDict.get(
+            PARAM_HVAC_MODE
+        )
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        value = self.coordinator.data[self.device.id].state.get_parameter_value(
+            PARAM_CURRENT_TEMPERATURE
+        )
+        if value is None:
+            return None
+        return float(value.value) if value is not None else None
```

Comment:
```suggestion
        return float(value.value)
```
check for `None` already done above